src/pages/ReportDetailsPage.tsx

```diff
@@ -686,6 +723,20 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
                     danger
                     shouldEnableNewFocusManagement
                 />
+                <ConfirmModal
+                    title={translate('iou.unapproveReport')}
+                    isVisible={isUnapproveModalVisible}
+                    danger
+                    confirmText={translate('iou.unapproveReport')}
+                    onConfirm={() => {
+                        setIsUnapproveModalVisible(false);
+                        Navigation.dismissModal();
+                        IOU.unapproveExpenseReport(moneyRequestReport);
```

Comment:
NAB: We can move the `onConfirm` logic into a separate function.

Reply:
good call 👍

Reply:
wait why?

Reply:
For better readability of the code , it's just a preference 😄

Reply:
I assumed we usually wanted to extract this into a separate function, but i see that we do this decently often so let's keep it here